### PR TITLE
Release 2.7.0

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
...
 
+## 2.7.0
+### Added
+* CLI: Added `onefuzz containers files download_dir` to enable downloading the contents of a container.  [#598](https://github.com/microsoft/onefuzz/pull/598)
+* Agent: Added `microsoft_telemetry_key` and `instance_telemetry_key` and expanded the availability `reports_dir` in [configuration value expansion](docs/command-replacements.md). [#561](https://github.com/microsoft/onefuzz/pull/561)
+* Agent/Service: Added `job_id` to agent-based heartbeats. [#594](https://github.com/microsoft/onefuzz/pull/594)
+
+### Changed
+* Agent/Proxy/Supervisor: Renamed the Application Insights token names used for telemetry to `microsoft_telemetry_key` and `instance_telemetry_key` and the function that gated telemetry sharing to `can_share_with_microsoft` to make the telemetry implementation easier to understand. [#587](https://github.com/microsoft/onefuzz/pull/587)
+* Deployment: Updated multiple Python dependencies. [#596](https://github.com/microsoft/onefuzz/pull/426)
```
Comment (ranweiler) [inline CHANGELOG.md:15]: Link text/target mismatch. Based on the description, I'm not sure if you actually meant to link to PR 426.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
...
+* Service: Updated multiple Python dependencies. Addresses potential security issue [CVE-2020-28493](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-28493) [#595](https://github.com/microsoft/onefuzz/pull/595)
+* Service: No longer allows a node that is part of a scaleset or pool that is scheduled to be shut down. [#583](https://github.com/microsoft/onefuzz/pull/583)
```
Comment (ranweiler) [inline CHANGELOG.md:17]: I think this description may be inaccurate. Did you want something like:
```suggestion
* Service: Don't let nodes run new tasks if they are part of a scaleset or pool that is scheduled to be shut down. [#583](https://github.com/microsoft/onefuzz/pull/583)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
...
+
+### Fixed
+* Service: Fixed the queries used to identify nodes running outdated OneFuzz releases. [#597](https://github.com/microsoft/onefuzz/pull/597)
+* Agent: Fixed an issue that would stop an agent or supervisor from performing work in upon HTTPS request has failed in certain conditions. [#603](https://github.com/microsoft/onefuzz/pull/603)
```
Comment (ranweiler) [inline CHANGELOG.md:21]:
```suggestion
* Agent: Fixed an issue that would stop an agent or supervisor from performing work if an HTTPS request has failed in certain conditions. [#603](https://github.com/microsoft/onefuzz/pull/603)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
...
+* Agent: Fixed an issue that would stop a task if the task printed significant amount of data to STDOUT or STDERR.  [#588](https://github.com/microsoft/onefuzz/pull/588)
```
Comment (ranweiler) [inline CHANGELOG.md:22]:
```suggestion
* Agent: Fixed an issue that would stop a task if the task printed a significant amount of data to STDOUT or STDERR.  [#588](https://github.com/microsoft/onefuzz/pull/588)
```

**Review by ranweiler — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bmc-msft) [reply CHANGELOG.md:15]:
```suggestion
* Deployment: Updated multiple Python dependencies. [#596](https://github.com/microsoft/onefuzz/pull/596)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 2.7.0
+### Added
+* CLI: Added `onefuzz containers files download_dir` to enable downloading the contents of a container.  [#598](https://github.com/microsoft/onefuzz/pull/598)
+* Agent: Added `microsoft_telemetry_key` and `instance_telemetry_key` and expanded the availability `reports_dir` in [configuration value expansion](docs/command-replacements.md). [#561](https://github.com/microsoft/onefuzz/pull/561)
+* Agent/Service: Added `job_id` to agent-based heartbeats. [#594](https://github.com/microsoft/onefuzz/pull/594)
+* Agent/Proxy/Supervisor: Added additional context to error conditions during Storage Queue interactions and service interactions to improve debugging.  [#601](https://github.com/microsoft/onefuzz/pull/601)
```
Comment (ranweiler) [inline CHANGELOG.md:12]:
```suggestion
* Agent/Proxy/Supervisor: Added additional context to errors during Storage Queue and service interactions to improve debugging.  [#601](https://github.com/microsoft/onefuzz/pull/601)
```

**Review by ranweiler — APPROVED** [review]

**Review by ranweiler — APPROVED** [review]

**Pull request merged**